lib/timers.js

```diff
@@ -626,9 +636,6 @@ exports.clearImmediate = function(immediate) {
 
   immediate._onImmediate = undefined;
 
-  L.remove(immediate);
-
-  if (L.isEmpty(immediateQueue)) {
-    process._needImmediateCallback = false;
-  }
+  // leave queued, much faster overall to skip it later in processImmediate
```

Comment:
> much faster

I don't think this is true, `L.remove()` is a constant-time operation. The performance gained will be minimal at best, insignificant either way.